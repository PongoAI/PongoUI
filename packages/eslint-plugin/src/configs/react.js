/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
