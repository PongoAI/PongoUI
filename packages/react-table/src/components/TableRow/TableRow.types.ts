import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableRowSlots = {
  /**
   * The root of the TableRow.
   */
  root: Slot<'tr'>;
};

type TableRowCommons = {};

export type TableRowProps = ComponentProps<TableRowSlots> & TableRowCommons;

export type TableRowState = ComponentState<TableRowSlots> & TableRowCommons;
