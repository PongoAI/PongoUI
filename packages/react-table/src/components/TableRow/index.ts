export * from './TableRow';
export * from './useTableRowStyles';
export * from './TableRow.types';
export * from './useTableRow';
