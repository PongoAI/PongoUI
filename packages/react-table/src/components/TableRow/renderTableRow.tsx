import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableRowState, TableRowSlots } from './TableRow.types';

export const renderTableRow = (state: TableRowState) => {
  const { slots, slotProps } = getSlots<TableRowSlots>(state);

  return <slots.root {...slotProps.root} />;
};
