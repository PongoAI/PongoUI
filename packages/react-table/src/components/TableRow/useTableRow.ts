import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableRowProps, TableRowState } from './TableRow.types';

export const useTableRow = (props: TableRowProps, ref: React.Ref<HTMLElement>): TableRowState => {
  const state: TableRowState = {
    components: {
      root: 'tr',
    },
    root: getNativeElementProps('tr', {
      ref,
      ...props,
    }),
  };

  return state;
};
