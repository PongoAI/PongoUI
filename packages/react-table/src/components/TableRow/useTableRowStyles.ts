import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { TableRowState } from './TableRow.types';

const useRootStyles = makeStyles({
  root: {
    display: 'table-row',
    verticalAlign: 'inherit',
    ...shorthands.borderColor('inherit'),
  },
});

export const useTableRowStyles = (state: TableRowState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
