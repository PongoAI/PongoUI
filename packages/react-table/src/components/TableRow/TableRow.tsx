import * as React from 'react';
import { useTableRow } from './useTableRow';
import { useTableRowStyles } from './useTableRowStyles';
import { renderTableRow } from './renderTableRow';
import type { TableRowProps } from './TableRow.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The TableRow component is used to display a row of TableCell's.
 */
export const TableRow: ForwardRefComponent<TableRowProps> = React.forwardRef((props, ref) => {
  const state = useTableRow(props, ref);

  useTableRowStyles(state);

  return renderTableRow(state);
});
TableRow.displayName = 'TableRow';
