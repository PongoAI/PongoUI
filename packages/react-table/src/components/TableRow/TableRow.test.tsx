import * as React from 'react';
import { TableRow } from './TableRow';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('TableRow', () => {
  isConformant({
    Component: TableRow,
    displayName: 'TableRow',
  });

  it('renders a basic Table', () => {
    const renderedComponent = render(<TableRow />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
