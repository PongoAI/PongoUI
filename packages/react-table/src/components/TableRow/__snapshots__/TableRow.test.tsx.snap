// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders a basic Table 1`] = `
<body>
  <div>
    <tr
      class="___5s2gsu0 f1u0rzck fvc3tfl fetgrfm fwsgm2r f1kc6foo f13jenf0"
    />
  </div>
</body>
`;
